[SecuritySolution][Alerts table] Fix issue with multiple ip addresses in strings

janmonschke opened the pull request:
## Summary

Fixes https://github.com/elastic/kibana/issues/191767

Multiple IPs are now displayed as individual links, even in the case where multiple IPs are passed as a single string (e.g. `127.0.0.1,127.0.0.2`). Clicking on an individual link will open the flyout correctly as well.


https://github.com/user-attachments/assets/74b05cff-3843-4149-bf27-cd0af07aa558



### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios


<!--ONMERGE {"backportTargets":["8.16","8.17","8.18","8.x","9.0"]} ONMERGE-->

janmonschke committed:
fix issue with multiple ip addresses in strings

elasticmachine commented:
Pinging @elastic/security-threat-hunting-investigations (Team:Threat Hunting:Investigations)

PhilippeOberti reviewed (approved):
Nice fix, code LGTM and desk tested! While we need to improve the UI to better show the ellipsis and the values in the tooltip in both the alerts table and timeline (and probably other places as well), this is outside of this PR's scope.
Nice job, thanks for fixing this!

janmonschke committed:
remove unused `<Content />` wrapper

The content wrapper isn't need anymore and it prevented the ellipsis from working

janmonschke commented:
> While we need to improve the UI to better show the ellipsis and the values in the tooltip in both the alerts table and timeline (...)

@PhilippeOberti Made the ellipsis work in https://github.com/elastic/kibana/pull/209475/commits/a9bc4ce323d364d1dbb98b93a72ba76fbc550bc6

janmonschke commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into security/multiple-ip-links

PhilippeOberti commented:
> > While we need to improve the UI to better show the ellipsis and the values in the tooltip in both the alerts table and timeline (...)
> 
> @PhilippeOberti Made the ellipsis work in [a9bc4ce](https://github.com/elastic/kibana/commit/a9bc4ce323d364d1dbb98b93a72ba76fbc550bc6)

Tested! Works beautifully, nice job! We lost the tooltip but there was not any useful information in it anyway as it was not displaying the value... plus I checked and the user name, host name, rule name (basically the other links) don't have it either.
Awesome!

janmonschke commented:
> We lost the tooltip but there was not any useful information (...)

Yeah, I should've specified that the wrapper wasn't used anymore and it was actually showing a tooltip when it was not used anywhere else in the table.

kapral18 commented on x-pack/solutions/security/plugins/security_solution/public/common/components/links/index.tsx:
```diff
@@ -298,58 +298,81 @@ export interface NetworkDetailsLinkProps {
-  ip,
-  flowTarget = FlowTarget.source,
+const NetworkDetailsLinkComponent: React.FC<NetworkDetailsLinkProps> = ({ ip, ...restProps }) => {
+  // We see that sometimes the `ip` is passed as a string value of "IP1,IP2".
```
can the same happen for arrays, i.e. `["IP1,IP2", "IP3,IP4"]`?

janmonschke replied:
I'm not sure to be honest. It's not something that I've seen before though.

PhilippeOberti replied:
I'd say, let's not make the code more complex. I don't think an array of string with multiple values like that is a situation that we need to handle...

kapral18 reviewed (approved):
LGTM with 1 questions, thanks!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/273597)
* Commit: 737cf97c5d1d7701ea410175693074bb6de48708

### Failed CI Steps
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/273597#0194d6a4-7221-412d-b440-0020a1a562aa)
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/273597#0194d6c7-6a9c-4c7f-9dd2-65856233890d)
* [FTR Configs #42](https://buildkite.com/elastic/kibana-pull-request/builds/273597#0194d6a4-7255-4a54-9795-0c8876bf7bec)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/273597#0194d6a4-7221-412d-b440-0020a1a562aa) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/273597/jobs/0194d6a4-7221-412d-b440-0020a1a562aa/artifacts/0194d6c7-4e17-4b63-8aa6-98be7459fdd8) Jest Tests #<span></span>13 / Netflow renders correctly against snapshot
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/273597#0194d6c7-6a9c-4c7f-9dd2-65856233890d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/273597/jobs/0194d6c7-6a9c-4c7f-9dd2-65856233890d/artifacts/0194d6e4-8348-422b-a2c1-4bb0b90b2bde) Jest Tests #<span></span>13 / Netflow renders correctly against snapshot
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/273597#0194d6a4-7221-412d-b440-0020a1a562aa) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/273597/jobs/0194d6a4-7221-412d-b440-0020a1a562aa/artifacts/0194d6c7-4e16-45e7-b6a5-4e0dc0f2c5ea) Jest Tests #<span></span>13 / netflowRowRenderer renders correctly against snapshot
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/273597#0194d6c7-6a9c-4c7f-9dd2-65856233890d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/273597/jobs/0194d6c7-6a9c-4c7f-9dd2-65856233890d/artifacts/0194d6e4-8349-4e6e-a9bc-fd2d1b52a59f) Jest Tests #<span></span>13 / netflowRowRenderer renders correctly against snapshot
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/273597#0194d6a4-7255-4a54-9795-0c8876bf7bec) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/273597/jobs/0194d6a4-7255-4a54-9795-0c8876bf7bec/artifacts/0194d6c3-d159-4026-881a-47a42d4d062d) FTR Configs #<span></span>42 / Rules Management - Prebuilt Rules - Prebuilt Rules Management @<span></span>ess @<span></span>serverless @<span></span>skipInServerlessMKI Bootstrap Prebuilt Rules should install fleet packages required for detection engine to function

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/273597","number":273597,"commit":"737cf97c5d1d7701ea410175693074bb6de48708"}],"number":273597}
buildkite-pr-comment-->

janmonschke committed:
update snapshots

kibanamachine commented:
Starting backport for target branches: 8.16, 8.17, 8.18, 8.x, 9.0

https://github.com/elastic/kibana/actions/runs/13166188116

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/273736)
* Commit: d41e376d3e00178539cf96fa74e52f7cf4fe3dd7

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/8a26cf6f94980008efaa169ab9150e4a40c94914) | [after](https://github.com/elastic/kibana/commit/d41e376d3e00178539cf96fa74e52f7cf4fe3dd7) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 21.5MB | 21.5MB | +225.0B |



### History
* :broken_heart: [Build #273597](https://buildkite.com/elastic/kibana-pull-request/builds/273597) failed 737cf97c5d1d7701ea410175693074bb6de48708

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/273736","number":273736,"commit":"d41e376d3e00178539cf96fa74e52f7cf4fe3dd7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/273597","number":273597,"commit":"737cf97c5d1d7701ea410175693074bb6de48708"}],"number":273736}
buildkite-pr-comment-->

kibanamachine commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.16|Backport failed because of merge conflicts|
|❌|8.17|Backport failed because of merge conflicts|
|✅|8.18|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/209907">](https://github.com/elastic/kibana/pull/209907)|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/209908">](https://github.com/elastic/kibana/pull/209908)|
|✅|9.0|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/209909">](https://github.com/elastic/kibana/pull/209909)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 209475
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->

janmonschke commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.17|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/209971">](https://github.com/elastic/kibana/pull/209971)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.4 -->

mistic commented:
This PR didn't make it into the latest BC for v8.17.2. Updating the labels.
